Documentation/ContinousIntegration.md

```diff
@@ -0,0 +1,42 @@
+# Building Swift Packages or Apps that Use Them in Continuous Integration Workflows
+
+Build Swift packages with an existing continuous integration setup and prepare apps that consume package dependencies within an existing CI pipeline.
+
+## Overview
+
+*Continuous integration* (*CI*) is the process of automating and streamlining the building, analyzing, testing, archiving, and publishing of your apps to ensure that they're always in a releasable state.
+Most projects that contain or depend on Swift packages don't require additional configuration.
+
+## Use the Expected Version of a Package Dependency
+
+To ensure the *CI* workflow’s reliability, make sure it uses the appropriate version of package dependencies.
+SwiftPM records the result of dependency resolution in `Package.resolved` (at the top-level of the package) and it's used when performing dependency resolution (rather than having SwiftPM searching the latest eligible version of each package).  Running `swift package update` updates all dependencies to the latest eligible versions and updates the `Package.resolved`.  You can commit `Package.resolved` to your *Git* repository to ensure it’s always up-to-date on the *CI* environment to prevent the *CI* from building your project with unexpected versions of package dependencies.  Otherwise you can choose to add `Package.resolved` file to `.gitignore` file and have `swift package resolve` command in charge of resolving the dependencies (`swift package resolve` is invoked by most SwiftPM commands).
+
+## Provide Credentials
```

Comment:
I am not sure about this one. I think we should explain here that SwiftPM relies on git for authentication, and as such the CI machines need to be configure so that git has access to the credentials it needs to access the repositories.

paraphrasing from https://developer.apple.com/documentation/swift_packages/building_swift_packages_or_apps_that_use_them_in_continuous_integration_workflows:

```
[To access private repositories] use SSH–based Git URLs for your packages and configure your SSH credentials. Set up your known_hosts file in the ~/.ssh directory of the user that runs your CI tasks. SwiftPM honors your SSH configuration — there’s no additional setup required.
If your SSH keys are password-protected, add them to the SSH agent before invoking SwiftPM by modifying the SSH configuration file.
````

We should also mention that CI services like Jenkins, Github Action, TravisCI, CircleCI, etc all provide ways to set up credentials and maybe link to their guides

Finally we need to mention in a new section that users can also use `xcodebuild` when building on macOS based CI hosts. 

cc @neonichu

Reply:
Yep, I agree. I think we should only have a generic description of the process here and link off to somewhere else for specifics since the setup can be different depending on the CI solution.